### ISSUE TITLE
APE file not supported, MAC codecs installed

### ISSUE THREAD

**Issue by Vash63** [flacon/flacon#201]
When I try to open an APE file with 'Add disc' with Flacon I get the following:

![image](https://user-images.githubusercontent.com/3086017/235291659-2ffe26f7-bb81-47da-9ebd-94b13f0677b8.png)

I have monkey's audio 10.08 installed to `/usr/bin/mac` via the Arch package, and the audio plays without issue in both mpv and vlc. Settings in the app are:

![image](https://user-images.githubusercontent.com/3086017/235291740-6a158fbc-c4f7-4554-9af4-69448018f104.png)

Does something else need to be done for APE files to be supported? A matching .cue is in the same directory.

**Comment by SokoloffA** [comment]
Please run the program from the console with debugging and send me the full output.
```flacon --debug 2>&1 | tee flacon.log```

Note that playing the file through a video player does not guarantee the integrity of the file. The player's should to play as many files as possible, so it may skip corrupted parts of file. Audio codecs are much more strict in this matter. You can check the file by decoding it through mac by hand.
```mac YOUR_APE_FILE.ape - -d > out.wav```

**Comment by Vash63** [comment]
[flacon.log](https://github.com/flacon/flacon/files/11371548/flacon.log)

And the output from `mac` : 
```
mac Nirvana\ -\ 1991\ -\ Nevermind.ape - -d > blah.wav
--- Monkey's Audio Console Front End (v 10.08) (c) Matthew T. Ashland ---
Decompressing...
Progress: 100.0% (0.0 seconds remaining, 18.2 seconds total)          
Success...
```
Maybe it doesn't support reading from my smb share? The file chooser lets me select it and the CUE file works from the same directory.

**Comment by SokoloffA** [comment]
So far I have not been able to reproduce the problem. Next week I will be on vacation, I will try to reproduce the problem.
Please give me more information:
1. Are you using the flatpack version of the program or from the package?
2. Which desktop environment do you use?
3. How did you choose samba shara? Through the "Other Locations" button or how? Please describe the process step by step, as for an idiot.
4. What does your phrase _" the CUE file works from the same directory."_ mean? I.e. if you select CUE file, the program  successfully load CUE+Audio, but if you select audio, you get an error. So?
5. And also, did you run the program with or without the "--debug" key?

**Comment by Vash63** [comment]
> So far I have not been able to reproduce the problem. Next week I will be on vacation, I will try to reproduce the problem. Please give me more information:
> 
>     1. Are you using the flatpack version of the program or from the package?
From package
>     2. Which desktop environment do you use?
Gnome 44
>     3. How did you choose samba shara? Through the "Other Locations" button or how? Please describe the process step by step, as for an idiot.
From Files/Nautilus, Other Locations, navigate to it, right click -> open in terminal, `pwd` and copy the path.
>     4. What does your phrase _" the CUE file works from the same directory."_ mean? I.e. if you select CUE file, the program  successfully load CUE+Audio, but if you select audio, you get an error. So?
If I select the CUE, the CUE itself opens fine and album details are loaded, but it still gives an error that audio was not found.
>     5. And also, did you run the program with or without the "--debug" key?
The earlier output was including the `--debug` flag

**Comment by SokoloffA** [comment]
@Vash63 

I have not been able to reproduce the bug in the wild, it depends on many factors, such as network and disk speed. But I wrote a fake audio decoder that returns bytes with delays. The problem was reproduced with it.

I fixed the bug, I hope. I will be happy if you tests my fixes using [AppImage](https://github.com/flacon/flacon/suites/12726686244/artifacts/683632544).

If you have any problems with launching AppImage, write, unfortunately I don't have Linux at hand, and I haven't checked the launch of AppImage.
